psi4/driver/procrouting/dft/xdm_functionals.py

```diff
@@ -0,0 +1,283 @@
+    }
+})
+funcs.append({
+    "name": "BHANDHLYP-XDM",
+    "xc_functionals": {
+        "HYB_GGA_XC_BHANDHLYP": {}
+    },
+    "citation":
+    '    A. D. Becke and E. R. Johnson, J. Chem. Phys. 127, 154108 (2007).\n' + \
+    '    A. Otero-de-la Roza, E. R. Johnson, J. Chem. Phys. 138, 204109 (2013).\n',
+    "description":
+    '    BHANDHLYP Exchange-Correlation Functional plus XDM dispersion (hybrid).\n',
+    "dispersion": {
+        "type": "xdm",
+        "params": {
+            'a1': 0.5610,
+            'a2': 1.9894,
+            'vol': 'bhah',
+        }
+    }
+})
+funcs.append({
+    "name": "TPSS-XDM",
+    "x_functionals": {
+        "MGGA_X_TPSS": {}
+    },
+    "c_functionals": {
+        "MGGA_C_TPSS": {}
+    },
+    "citation":
+    '    A. D. Becke and E. R. Johnson, J. Chem. Phys. 127, 154108 (2007).\n' + \
+    '    A. Otero-de-la Roza, E. R. Johnson, J. Chem. Phys. 138, 204109 (2013).\n',
+    "description":
+    '    TPSS Exchange-Correlation Functional plus XDM dispersion (meta-GGA).\n',
+    "dispersion": {
+        "type": "xdm",
+        "params": {
+            'a1': 0.6612,
+            'a2': 1.5111,
+            'vol': '0.0',
+        }
+    }
+})
+funcs.append({
+    "name": "HSE06-XDM",
+    "xc_functionals": {
+        "HYB_GGA_XC_HSE06": {}
+    },
+    "citation":
+    '    A. D. Becke and E. R. Johnson, J. Chem. Phys. 127, 154108 (2007).\n' + \
+    '    A. Otero-de-la Roza, E. R. Johnson, J. Chem. Phys. 138, 204109 (2013).\n',
+    "description":
+    '    HSE06 Exchange-Correlation Functional plus XDM dispersion (rs-hybrid).\n',
+    "dispersion": {
+        "type": "xdm",
+        "params": {
+            'a1': 0.3691,
+            'a2': 2.8793,
+            'vol': '0.0',
+        }
+    }
+})
+funcs.append({
+    "name": "BP86-XDM",
+    "x_functionals": {
+        "GGA_X_B88": {}
+    },
+    "c_functionals": {
+        "GGA_C_P86": {}
+    },
+    "citation":
+    '    A. D. Becke and E. R. Johnson, J. Chem. Phys. 127, 154108 (2007).\n' + \
+    '    A. Otero-de-la Roza, E. R. Johnson, J. Chem. Phys. 138, 204109 (2013).\n',
+    "description":
+    '    BP86 Exchange-Correlation Functional plus XDM dispersion (GGA).\n',
+    "dispersion": {
+        "type": "xdm",
+        "params": {
+            'a1': 0.8538,
+            'a2': 0.6415,
+            'vol': '0.0',
+        }
+    }
+})
+funcs.append({
+    "name": "B86bPBE-XDM",
+    "x_functionals": {
+        "GGA_X_B86_MGC": {}
+    },
+    "c_functionals": {
+        "GGA_C_PBE": {}
+    },
+    "citation":
+    '    A. D. Becke and E. R. Johnson, J. Chem. Phys. 127, 154108 (2007).\n' + \
+    '    A. Otero-de-la Roza, E. R. Johnson, J. Chem. Phys. 138, 204109 (2013).\n',
+    "description":
+    '    B86bPBE Exchange-Correlation Functional plus XDM dispersion (GGA).\n',
+    "dispersion": {
+        "type": "xdm",
+        "params": {
+            'a1': 0.7839,
+            'a2': 1.2544,
+            'vol': '0.0',
+        }
+    }
+})
```

Comment:
IIRC XDM is not done self-consistently. Could we try to add this more likely canonical -D terms where we treat it as a post-processing step.

Reply:
In the beginning (in q-chem at least), XDM was done self-consistently. But I think they decided it didn't matter.

I was thinking of setting it up as dftd3-like (fall back on nl-like) as possible. Most unwieldy item seems to be these `vol`s where there are a periodic table's-worth for each of 9 fctls (https://github.com/aoterodelaroza/postg/blob/master/param.f90#L220-L317), then a fallback to %HF scale of a generic data block. So that `vol` really is a string or float.

Reply:
The XDM implementation in q-chem is indeed self-consistent but, as you point out, it does not matter very much (or at all) and post-SCF is much simpler. The atomic volumes are there because optimal results are obtained when the atom-in-molecule volume calculated from your system goes to the "correct" in-vacuo atomic volume if the molecule is taken apart into atoms. If the functional in question is not known, then the fraction of exact exchange is used instead, which more or less solves the problem.